### ISSUE TITLE
sockets: filtering on fabric name doesn't seem to work as expected

### ISSUE THREAD

**Issue by bturrubiates** [ofiwg/libfabric#2248]
In trying to debug an unrelated issue, I noticed some unexpected behavior with the sockets provider. The code for this test can be found [here](https://github.com/bturrubiates/fi-resolve).

It's a very simple program to just try resolving an address. I'd expect that if I request fabric `127.0.0.0/8`, then I'll only get back structures from `fi_getinfo` that are on that fabric. 

The `-P` switch is the provider, the `-f` switch is the fabric, and the last parameter is the address that should be resolved. 

```sh
bturrubi at ivy11 :: ./src/fi_resolve -P sockets -f 127.0.0.0/8 10.30.0.191
[src/resolve.c]:setup_fabric():<91> provided provider name hint: sockets
[src/resolve.c]:setup_fabric():<92> provided fabric name hint: 127.0.0.0/8
[src/resolve.c]:setup_fabric():<111> opened fabric with provider: sockets
[src/resolve.c]:setup_fabric():<112> bound to: 10.0.0.0/16
Successfully resolved address to 10.30.0.191:1337
```

**Comment by shantonu** [comment]
Ok, I'll check <code>fi_getinfo</code> logic. Thanks for the reproducer :)

**Comment by shantonu** [comment]
With fabric name filtering I get the following output-

```
fi_resolve -P sockets -f 127.0.0.0/8 10.30.0.191
[src/resolve.c]:setup_fabric():<91> provided provider name hint: sockets
[src/resolve.c]:setup_fabric():<92> provided fabric name hint: 127.0.0.0/8
[src/resolve.c]:setup_fabric():<111> opened fabric with provider: sockets
[src/resolve.c]:setup_fabric():<112> bound to: 127.0.0.0/8
Successfully resolved address to 10.30.0.191:1337
```

It should still be able to resolve the address, right? @bturrubiates

**Comment by bturrubiates** [comment]
> With fabric name filtering I get the following output-
> 
> fi_resolve -P sockets -f 127.0.0.0/8 10.30.0.191
> [src/resolve.c]:setup_fabric():<91> provided provider name hint: sockets
> [src/resolve.c]:setup_fabric():<92> provided fabric name hint: 127.0.0.0/8
> [src/resolve.c]:setup_fabric():<111> opened fabric with provider: sockets
> [src/resolve.c]:setup_fabric():<112> bound to: 127.0.0.0/8
> Successfully resolved address to 10.30.0.191:1337
> It should still be able to resolve the address, right? @bturrubiates

I think this looks correct. Alternatively, you could also play around with the `-f` argument for the changes I made in #2300. 

The resolvable portion is just an `fi_av_insert`. So for usNIC if an address is not resolvable or reachable, then the insert fails.
